### DISCUSSION TITLE
Spawning items in range weapons doesn't apply rotation correctly

### DISCUSSION THREAD

**Discussion by kleiner-sentry** [FakeFishGames/Barotrauma#13680] [Code questions and modding support]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

I'm trying to make a spear that you can throw, and when thrown it'll deal damage but also attach to walls and characters. I could make the item a <Throwable>, however that wont give me the desired effects. So, I made the item into a <RangedWeapon> and a <Projectile>, so when the user uses the spear it'll remove itself and spawn itself again, but this time as a projectile that's launched at high speed, deals damage, and stick to walls. There's only one problem. It's able to copy the position (I think), but it doesn't seem to copy the rotation despite all of the code should theoretically copy the rotation of the item. 

Here's the item:
![image](https://github.com/FakeFishGames/Barotrauma/assets/5957082/93f7a912-e904-46a0-aa52-c2fa9f340d0e)
[Scugitems.zip](https://github.com/FakeFishGames/Barotrauma/files/14358044/Scugitems.zip)

### Reproduction steps

1. Create a custom item
2. Make the item a ranged weapon, and a projectile
3. make the item spawn itself when you use the weapon
4. bug

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.2.8.0 (Winter Update hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

Looking at the code, you seem to be using `rotationtype="Limb"` to spawn the second spear. I suspect that should be `rotationtype="This"` (referring to the first spear).

**Reply by kleiner-sentry** [reply]
if i remember correctly, I'm pretty sure I used "this" in the spear, and it still didn't work.
